source/class/qx/ui/treevirtual/MNode.js

```diff
@@ -173,11 +199,16 @@ qx.Mixin.define("qx.ui.treevirtual.MNode", {
      *   represented either by the node object, or the node id (as would have
      *   been returned by addBranch(), addLeaf(), etc.)
      *
-     * @param b {Boolean}
+     * @param opened {Boolean}
      *   The new opened state for the specified node.
      *
+     * @param cascade {Boolean}
+     *   Whether to descend the tree changing opened state of all children
+     *
+     * @param isRecursed {Boolean?}
+     *   For internal use when cascading to determine outer level and call setData
      */
-    nodeSetOpened(nodeReference, b) {
+    nodeSetOpened(nodeReference, opened, cascade, isRecursed) {
```

Comment:
I see the need for `isRecursed` but it seems ugly to expose that to the user. What do you think about having `nodeSetOpened` call an internal method, e.g., `_nodeSetOpenedInternal`, that adds/uses that additional argument?

Reply:
Added private method